src/client/opamSolution.ml

```diff
@@ -824,6 +825,90 @@ let run_hook_job t name ?(local=[]) w =
   | None ->
     Done true
 
+let rec check_and_install_depexts t packages =
+  let sys_packages =
+    if OpamStateConfig.(!r.depext_ignore_opam) then
+      OpamSwitchState.system_packages t.switch_config packages
+        ~depexts:(OpamSwitchState.depexts t)
+    else t.sys_packages
+  in
+  let avail, nf =
+    let add pkg sys_set set =
+      if not (OpamSysPkg.Set.is_empty sys_set) then
+        OpamPackage.Map.add pkg
+          (OpamSysPkg.Set.elements sys_set) set
+      else set
+    in
+    OpamPackage.Map.fold (fun pkg sys (avail,nf) ->
+        if OpamPackage.Set.mem pkg packages then
+          add pkg sys.OpamSysPkg.s_available avail,
+          add pkg sys.OpamSysPkg.s_not_found nf
+        else
+          avail,nf)
+      sys_packages (OpamPackage.Map.empty, OpamPackage.Map.empty)
+  in
+  let print_dep pkgs msg =
+    let lst =
+      OpamPackage.Map.bindings
+        (OpamPackage.Map.map (List.map OpamSysPkg.to_string) pkgs)
+    in
+    if lst = [] then "" else
+      OpamStd.Format.itemize (fun lst ->
+          Printf.sprintf "%s\n%s"
+            msg
+            (OpamStd.Format.itemize (fun (p,s) ->
+                 Printf.sprintf "%s: %s"
+                   (OpamPackage.to_string p)
+                   (OpamStd.Format.pretty_list s))
+                lst)
+        ) [lst]
+  in
+  if not (OpamPackage.Map.is_empty avail && OpamPackage.Map.is_empty nf) then
+    (OpamConsole.error "Some external dependencies are missing:\n%s%s"
+       (print_dep avail "available to install:")
+       (print_dep nf
+          (Printf.sprintf "not available on %s:"
+             (match OpamSysPoll.os_family () with
+                Some o -> o | None -> assert false)
+          ));
+     if not (OpamPackage.Map.is_empty avail
+             || OpamStateConfig.(!r.depext_not_root)) then
+       let packages =
+         OpamPackage.Map.values avail
+         |> List.flatten
+         |> OpamSysPkg.Set.of_list
+       in
+       if OpamStateConfig.(!r.depext_no_sudo && not !r.depext_not_root) then
+         let commands =
+           OpamSysInteract.install_packages_commands packages
+         in
+         OpamConsole.msg "Run %s command%s to install missing packages:\n%s\n"
+           (if List.length commands = 1 then "this" else "these")
+           (if List.length commands = 1 then "" else "s")
+           (OpamStd.Format.itemize ~bullet:"  "
+              (OpamStd.List.concat_map " " (fun x -> x))
+              commands)
```

Comment:
The idea behind `no_sudo` (renamed `print-only`) is to avoid unnecessary interactions. If user want to know and choose if opam continue (then pause it) or not is the default.